Cookbook/Proposals/Danger-Integration.md

```diff
@@ -0,0 +1,152 @@
+We think that those potential future rules (based on test results) are not our top priorities for now though and it's too early to decide to move Danger so late in the build pipeline, as it would mean only having feedback fron Danger _after_ the `test_pr` job has finished – which is usually one of the last to finish on PRs. It's totally imaginable to move the `danger` step after the test job in a later PR if it becomes a limitation for a rule we want to implement in the future.
+
+### Picking a different first rule
+
+There are other ideas for initial rules to include in our `Dangerfile`, including but not limiting to:
+
+* Analyse the `pbxproj` files to find missing references and potential `(null)` values and warn about them
+* Analyse the `pbxproj` files to detect any files that was added to a target but is typically one to exclude from the final bundle – examples including `Info.plist` files but also `Snapshot` files
+* Generate an informative comment if any of the build config changed in a project, so that reviewers would not miss those changes in a hard-to-read pbxproj diff
+* Generate an informative comment if the `Podfile.lock` changed, so that reviewers take extra care when pods are added, removed or updated on the project
+* Warn if the `CHANGELOG.md` hasn't been updated, and/or enforcing updating the SDK changelog if there was any change in `SDK/*`
+* Warn if the PR doesn't have the author assigned to it
+* Report SwiftLint warnings in the PR (though it would require to move the execution of Danger after the `test_pr` job, and besides, as of today those SwiftLint violations are reported as errors anyway)
+* ...
+
+Those rules could also have been picked as initial candidate for the first rule we implement in Danger, but the rule about the PR size seems more straitforward to implement as a good way to test the initial usage of Danger on our process.
+
+All the rules mentionned in the list above will get their own separate proposal, so that we can discuss their benefits, the exact conditions to trigger each rule, and alternatives considered for each of them separately.
+
+_Note: I am hoping to write proposals for some of those additional rules soon, so that we could discuss them altogether in the upcoming proposal meeting alongside this very proposal for Danger, if time allows. I'll update the list above with links as I publish those other proposals_
+
+### Different conditions for PR size Rule
+
+The rule propose to make the condition based on the "overall lines of code added/removed in the diff" (that's the value returned by `git.lines_of_code` as documented in Danger docs). Which means it checks the value of `insertions` + `deletions`. That means that a PR which adds 20 lines and remove 30 lines will be seen as a size of 50.
```

Comment:
This was something that was not very clear for me in our latest discussions and it might be that other team members thought about different way of applying this limit (so far exceeding the limit was raised on demand when it PR clearly too big), so definitely worth fixing it in a rule.

Another alternative could be to assert on additions and deletions separately, where each of them should not exceed the limit and so `+800/-800` is the maximum size of the PR. That was my understanding of the limit so far.

Reply:
* [ ] Add alternative to count `insertions` and `deletions` separately